lib/list.js

```diff
@@ -93,8 +82,24 @@ util.inherits(List, Array);
 
 const _push = List.prototype.push;
 
+List.prototype.toItem = function(item) {
```

Comment:
IIUC, you are moving the definition of this prototype method outside of the constructor and there is no change in functionality. Can you please extract this change into a standalone commit? A new pull request would be even better - this change can be landed very quickly.

When I was reviewing the changes, I got confused about how is this move related to `__parent` reference and whether this method was changed or not as part of your fix. When the move is in a different commit, then the intentions are more clear and the fix is easier to review.

Reply:
Yes you are right, I saw that the modification of the prototype was in the wrong place and moved it as well. The change that is related  to this request, is that for the usage of the **parent** property in the **List** class as it existed, the reference to actual model is passed instead of the inner __data object

Reply:
> The change that is related to this request, is that for the usage of the parent property in the List class as it existed, the reference to actual model is passed instead of the inner __data object

Oh, that part is difficult to spot when reviewing the changes. One more reason why the move of `toItem` should be done in a standalone pull request.

Reply:
Would you mind bypassing this for now, because issuing a new PR, backporting it to 3.X etc will take lots of time? If you are not willing to accept it, I will remove it from this request

Reply:
I see your point. Can you at least put the change moving `toItem` into a standalone commit please?

Reply:
Yes for sure!